### ISSUE TITLE
ERROR

### ISSUE THREAD

**Issue by beratersari** [ibaiGorordo/ONNX-YOLOv7-Object-Detection#12]
``
Traceback (most recent call last):

  File "image_object_detection.py", line 15, in <module>

  boxes, scores, class_ids = yolov7_detector(img)

  File "/home/berat/berat-projects/ekin/ONNX-YOLOv7-Object-Detection/yolov7/YOLOv7.py", line 20, in __call__

  return self.detect_objects(image)

  File "/home/berat/berat-projects/ekin/ONNX-YOLOv7-Object-Detection/yolov7/YOLOv7.py", line 44, in detect_objects

  self.boxes, self.scores, self.class_ids = self.process_output(outputs)

  File "/home/berat/berat-projects/ekin/ONNX-YOLOv7-Object-Detection/yolov7/YOLOv7.py", line 78, in process_output

  predictions = predictions[obj_conf > self.conf_threshold]

IndexError: boolean index did not match indexed array along dimension 2; dimension is 80 but corresponding boolean 

dimension is 85
`

How can I fix that? I just tested the default yolov7.onnx

**Comment by ibaiGorordo** [comment]
Without more context it is hard to tell
- Where did you get yolov7.onnx?
- Does it have the nms/postprocessing included?
- Can you share the onnx model?
- What image/video are you testing it with?

**Comment by beratersari** [comment]
I fixed that. In my case, the output of onnx model has three outputs. Onnx models which you shared in the read.me file just concatenate these three outputs. I solved by adding some script to concatenate these outputs. You can see the last layer of my onnx model below.
![image](https://user-images.githubusercontent.com/56562278/205306573-6fa8dac7-235f-448d-a432-57ef4dad049a.png)

I also added these first seven lines to the YOLOV7.py file to concatenate the outputs shown in the image above. That function is in the 72 th line. 

```
def process_output(self, output):
        predictions0 = np.squeeze(output[0])
        predictions1 = np.squeeze(output[1])
        predictions2 = np.squeeze(output[2])
        
        predictions0=predictions0.reshape(-1, predictions0.shape[-1])
        predictions1=predictions1.reshape(-1, predictions1.shape[-1])
        predictions2=predictions2.reshape(-1, predictions2.shape[-1])
        
        predictions = np.concatenate((predictions0, predictions1,predictions2), axis=0)
        
        # Filter out object confidence scores below threshold
        obj_conf = predictions[:, 4]
        predictions = predictions[obj_conf > self.conf_threshold]
        
        obj_conf = obj_conf[obj_conf > self.conf_threshold]

        # Multiply class confidence with bounding box confidence
        predictions[:, 5:] *= obj_conf[:, np.newaxis]

        # Get the scores
        scores = np.max(predictions[:, 5:], axis=1)
        print(scores)
        # Filter out the objects with a low score
        predictions = predictions[scores > self.conf_threshold]
        scores = scores[scores > self.conf_threshold]
        print(predictions)
        print("predictions result",predictions.shape)
        print("scores result",scores.shape)

        if len(scores) == 0:
            return [], [], []

        # Get the class with the highest confidence
        class_ids = np.argmax(predictions[:, 5:], axis=1)

        # Get bounding boxes for each object
        boxes = self.extract_boxes(predictions)
        print(boxes)

        # Apply non-maxima suppression to suppress weak, overlapping bounding boxes
        indices = nms(boxes, scores, self.iou_threshold)
        print(indices)
        print(scores)
        return boxes[indices], scores[indices], class_ids[indices]

```
Now,I am not getting any bbox output from the visualizer. I guess there is a problem in my onnx file. Which onnx converter did you use to convert .pt file to .onnx file. I used the converter in the yolov7 repository (https://github.com/WongKinYiu/yolov7/blob/main/export.py) . But the last layer isn't like yours last layer. Also, I am getting warning while converting. If you share your onnx converter scirpt I will be very pleasure. Also, could you state onnxruntime version? Thanks in advance.

**Comment by beratersari** [comment]
https://drive.google.com/file/d/1m3y6dlm78J_WvVXHKuIARBkzqwBqRyFw/view?usp=sharing  (Coverted yolov7,pt file in yolov7 repository)
This onnx model is not same as the photo which I shared. However, I also could not get any bounding box output for this onnx model. It just shows the image without any bboxes. I also tried to use your yolov7-tiny_480x640.onnx model. It easily detects the cars in the photo. 

I tested it with this image
![1470848884_42a8](https://user-images.githubusercontent.com/56562278/205310644-82418165-6e2c-4b2c-9bc5-e253ed2a7c36.jpg)
